### PR TITLE
feat(api): add `ivr.proto`

### LINKED ISSUE #726: proposal: IVR API, or PhoneCall API
## What would you like to be added
IVR API, or PhoneCall API

Developers can invoke this API to send voice messages to specific people.

## Why is this needed
In the monitoring scenarios, monitor systems need to send alarm messages to people on-call. 
The messages might be in different forms, including IM,SMS, Email and phone calls, depending on the level of urgency.

## Product research
| IVR product |Docs|
|---|---|
|Aliyun VMS| https://www.aliyun.com/product/vms |
|AWS Pinpoint | https://aws.amazon.com/cn/pinpoint/ |

## Detailed Design

We need to consider the following factors:
- Portability
For example, a monitor system might be deployed on alibaba cloud(using [VMS](https://www.aliyun.com/product/vms) to send voice message) or AWS (using [AWS Pinpoint](https://aws.amazon.com/cn/pinpoint/)  to send voice message). So portability is important here.

```proto

service IvrService {

  //Send voice using the specific template
  rpc SendVoiceWithTemplate(SendVoiceWithTemplateRequest) returns (SendVoiceWithTemplateResponse) {}

}

message SendVoiceWithTemplateRequest{
  // Required
  Template template = 1;
  // Required
  string to_mobile = 2;
  // This field is required by some cloud providers.
  string from_mobile = 3;
}

message Template{
  // Required
  string template_id = 1;
  // Required
  map<string, string>  template_params = 2;
}

message SendVoiceWithTemplateResponse{
  string message_id = 1;
}


```

### PR DESCRIPTION
**Pull request by seeflood**
Signed-off-by: seeflood <zhou.qunli@foxmail.com>

**What this PR does**:
add `ivr.proto`

**Which issue(s) this PR fixes**:

Fixes #726 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: docs/en/design/notify/ivr.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+We need to consider the following factors:
+- Portability
+  For example, a monitor system might be deployed on alibaba cloud(using [VMS](https://www.aliyun.com/product/vms) to send voice message) or AWS (using [AWS Pinpoint](https://aws.amazon.com/cn/pinpoint/)  to send voice message). So portability is important here.
+
+```proto
+
+service IvrService {
+
+  //Send voice using the specific template
```
Comment (kevinten10) [inline docs/en/design/notify/ivr.md:29]: Haha, how about adding a space like：

`// Send voice using the specific template`

File: docs/en/design/notify/ivr.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  rpc SendVoiceWithTemplate(SendVoiceWithTemplateRequest) returns (SendVoiceWithTemplateResponse) {}
+
+}
+
+message SendVoiceWithTemplateRequest{
+  // Required
+  Template template = 1;
+  // Required
+  string to_mobile = 2;
```
Comment (kevinten10) [inline docs/en/design/notify/ivr.md:38]: How about multiple recipients?

File: spec/proto/runtime/v1/ivr.proto [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  //Send voice using the specific template
+  rpc SendVoiceWithTemplate(SendVoiceWithTemplateRequest) returns (SendVoiceWithTemplateResponse) {}
+
+}
+
+message SendVoiceWithTemplateRequest{
+  // Required
+  Template template = 1;
+  // Required
+  string to_mobile = 2;
```
Comment (kevinten10) [inline spec/proto/runtime/v1/ivr.proto]: repeated?

**Comment by ZLBer** [comment]
ivr is confusing  :-)

**Comment by seeflood** [comment]
> ivr is confusing :-)

Well, I can change it to `PhoneCall`

**Comment by seeflood** [comment]
> ivr is confusing :-)

@ZLBer fixed. Please review again

**Review by kevinten10 — APPROVED** [review]
LGTM.

**Review by zhenjunMa — APPROVED** [review]

**Review by wenxuwan — APPROVED** [review]

**Pull request merged**